pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1757,6 +1758,16 @@ public void deleteOfflineTable(String tableName) {
     HelixHelper.removeResourceFromBrokerIdealState(_helixZkManager, offlineTableName);
     LOGGER.info("Deleting table {}: Removed from broker resource", offlineTableName);
 
+    // Drop the table on servers
+    // TODO: Wait for externalview to converge on controllers instead of servers. This is because if externalview
```

Comment:
Can you also mention that we should make this API to be blocking and we also need to make this to be idempotent?

Reply:
Added